assets/model_monitoring/components/tests/e2e/conftest.py

```diff
@@ -609,3 +610,45 @@ def uploaded_version_file_name():
 def components_are_uploaded(uploaded_version_file_name):
     """Check if we already have a component version available uploaded by github CI."""
     return os.path.exists(uploaded_version_file_name)
+
+
+@pytest.fixture(scope="module")
+def submit_pipeline_job(ml_client: MLClient, request):
+    """Submit pipeline job to ml ws. Handle deletion if the testcase is cancelled/stopped."""
+    submitted_jobs = []
+
+    def _submit_job(job: Job, experiment_name: str):
+        pipeline_job = ml_client.jobs.create_or_update(
+            job, experiment_name=experiment_name, skip_validation=True
+        )
+        submitted_jobs.append(pipeline_job)
+        return pipeline_job
+
+    def cancel_jobs():
+        print("cancelling the jobs submitted:")
+        for job in submitted_jobs:
+            job_at_end = ml_client.jobs.get(job.name)
+            print(f"job: {job.name}, status: {job.status}, end_status: {job_at_end.status}")
+            if job_at_end.status not in RunHistoryConstants.TERMINAL_STATUSES:
+                ml_client.jobs.begin_cancel(job.name)
+
+    request.addfinalizer(cancel_jobs)
+    yield _submit_job
+
+
+@pytest.fixture(scope="session")
+def cleanup_previous_e2e_tests(ml_client: MLClient, test_suite_name):
```

Comment:
what's this test_suite_name? experiment name?

Reply:
test_suite_name is based on the "github_ref", test_suite_name we use as the experiment name. So when the tests run on a PR the github ref is for each PR so something like "merge_2346"